docs/source/json-api/index.rst

```diff
@@ -32,140 +32,123 @@ We welcome feedback about the JSON API on `our issue tracker
+
+Every authentication token is different and will depend on what your specific ledger or ledger operator requires.
+The JSON API server requires no access to party-specific data, only access to the ledger identity and package services.
+A token issued for the HTTP JSON API server should contain enough claims to contact these two services but no more than that.
+Please refer to your ledger operator's documentation to find out how.
+
+Once you have retrieved your access token, you can provide it to the JSON API by storing it in a file
+and starting ``daml json-api`` with the flag ``--access-token-file /path/to/your/token.file``.
+
+If the token cannot be read from the provided path or the Ledger API reports an authentication error
+(for example due to token expiration), the JSON API will report the error via logging. 
+
+.. note:: If the token file is updated with a new token it will be picked up at the next attempt to send a request. You can use this to handle cases where an old token expires without interrupting your JSON API service.
+
+Authorization
+-------------
+
+Every request from a client to the JSON API requires you to specify a party and some other settings,
+with a JWT token.  Normal HTTP requests pass the token in an ``Authorization`` header, 
+while WebSocket requests pass the token in a subprotocol.
+
+.. note:: While the JSON API receives the token it doesn't handle it itself. Upon receiving a token it will pass it on to the underlying Ledger's AuthService which will then determine if the token is valid and authorized. As such each token format is ledger/implementation-specific and the below example reflects only the format that is used for the DAML Sandbox.
+
+In the DAML Sandbox testing environment, you can use https://jwt.io to generate your
 token.  The default "header" is fine.  Under "Payload", fill in:
 
 .. code-block:: json
 
     {
       "https://daml.com/ledger-api": {
         "ledgerId": "MyLedger",
-        "applicationId": "foobar",
+        "applicationId": "HTTP-JSON-API-Gateway",
         "actAs": ["Alice"]
       }
     }
 
-Keep in mind that the value of the ``ledgerId`` payload field has to match the one passed to the sandbox with the ``--ledgerid`` argument. You can replace ``Alice`` with whatever party you want to use.
+The value of the ``ledgerId`` field has to match the ``ledgerId` of your underlying DAML Ledger.
+For the ``daml sandbox`` this corresponds to the ``--ledgerid MyLedger`` flag.
+
+The value of the ``applicationId`` field must match that provided to the ``daml json-api`` command with the flag
+``--application-id`` (default is ``HTTP-JSON-API-Gateway``).
+
+The value for ``actAs`` is specified as a list. You can replace ``Alice`` with whatever party you want to use,
+provided that it is a valid party on the ledger.
+For the ``daml sandbox`` any string will create a party with that name.
```

Comment:
Not quite. There are still restrictions on valid party names. And I think this is the wrong place to explain implicit party allocation anyway.

Reply:
Wouldn't `provided that it's a valid party on the ledger` cover `valid party names`?

I know as a user of the JSON API I would want to know how I can specify the party I want to use.

We should have a link to the section of the docs that explains implicit party allocation though.

Reply:
There are two cases:
1. The party is allocated on the ledger. Then it is clearly also a valid party string. This is what you should usually be working with on any ledger that doesn’t have implicit party allocation.
2. The party is not allocated but it is a valid party string. This only works if you have implicit party allocation. Otherwise, things will fail.